src/mono/mono/metadata/monitor.c

```diff
@@ -596,6 +606,46 @@ mono_object_hash_icall (MonoObjectHandle obj, MonoError* error)
 	return mono_object_hash_internal (MONO_HANDLE_RAW (obj));
 }
 
+int
+mono_object_try_get_hash_internal (MonoObject* obj)
+{
+#ifdef HAVE_MOVING_COLLECTOR
+
+	LockWord lw;
+	if (!obj)
+		return 0;
+	lw.sync = obj->synchronisation;
+
+	LOCK_DEBUG (g_message("%s: (%d) Get hash for object %p; LW = %p", __func__, mono_thread_info_get_small_id (), obj, obj->synchronisation));
+
+	if (lock_word_has_hash (lw)) {
+		if (lock_word_is_inflated (lw)) {
+			return lock_word_get_inflated_lock (lw)->hash_code;
+		} else {
+			return lock_word_get_hash (lw);
+		}
+	}
+
+	return 0;
+
+#else
+
+	unsigned int hash = HASH_OBJECT(obj);
+	if (hash == 0) {
+		hash = 1;
+	}
+	return hash;
+
+#endif
+
+}
+
+int
+mono_object_try_get_hash_icall (MonoObjectHandle obj, MonoError* error)
+{
```

Comment:
This would be better in object.c where the other object icalls are. Also, it should follow the
naming convention i.e. ves_icall_System_Object_ InternalTryGetHashCode.

Reply:
I renamed and moved these icalls to match the convention.